mod.sp

```diff
@@ -7,6 +7,76 @@ locals {
   }
 }
 
+variable "common_dimensions" {
+  type        = list(string)
+  description = "A list of common dimensions to add to each control."
+  # Define which common dimensions should be added to each control.
+  # - connection_name (_ctx ->> 'connection_name')
+  # - region
+  # - resource_group
+  # - subscription
+  # - subscription_id
+  default = ["resource_group", "subscription"]
+}
+
+variable "tag_dimensions" {
+  type        = list(string)
+  description = "A list of tags to add as dimensions to each control."
+  # A list of tag names to include as dimensions for resources that support
+  # tags (e.g. "department", "environment"). Default to empty since tag names are
+  # a personal choice
+  default     = []
+}
+
+locals {
+  # Local internal variable to build the SQL select clause for common
+  # dimensions using a table name qualifier if required. Do not edit directly.
+  common_dimensions_qualifier_sql = <<-EOQ
+  %{~if contains(var.common_dimensions, "connection_name")}, __QUALIFIER___ctx ->> 'connection_name' as connection_name%{endif~}
+  %{~if contains(var.common_dimensions, "region")}, __QUALIFIER__region as region%{endif~}
+  %{~if contains(var.common_dimensions, "resource_group")}, __QUALIFIER__resource_group as resource_group%{endif~}
+  %{~if contains(var.common_dimensions, "subscription_id")}, __QUALIFIER__subscription_id as subscription_id%{endif~}
+  EOQ
+
+  common_dimensions_global_qualifier_sql = <<-EOQ
+  %{~if contains(var.common_dimensions, "connection_name")}, __QUALIFIER___ctx ->> 'connection_name' as connection_name%{endif~}
+  %{~if contains(var.common_dimensions, "resource_group")}, __QUALIFIER__resource_group as resource_group%{endif~}
+  %{~if contains(var.common_dimensions, "subscription_id")}, __QUALIFIER__subscription_id as subscription_id%{endif~}
+  EOQ
+
+  common_dimensions_tenant_qualifier_sql = <<-EOQ
+  %{~if contains(var.common_dimensions, "connection_name")}, __QUALIFIER___ctx ->> 'connection_name' as connection_name%{endif~}
+  %{~if contains(var.common_dimensions, "tenant_id")}, __QUALIFIER__tenant_id as tenant_id%{endif~}
+  %{~if contains(var.common_dimensions, "subscription_id")}, __QUALIFIER__subscription_id as subscription_id%{endif~}
+  EOQ
+
+  common_dimensions_pricing_qualifier_sql = <<-EOQ
```

Comment:
can we rename this to `common_dimensions_subscription_id_qualifier_sql`?

Reply:
Yes we can rename `common_dimensions_pricing_qualifier_sql` to `common_dimensions_subscription_id_qualifier_sql`